src/clipboard.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use Clipboard;
+use SelectionData;
+use TargetEntry;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi::gpointer;
+use libc::c_uint;
+use std::boxed::Box as Box_;
+use std::mem::transmute;
+
+pub trait ClipboardExtManual {
+    fn set_with_data<F: Fn(&Clipboard, &SelectionData, c_uint) + 'static>(&self, targets: &[TargetEntry], f: F) -> bool;
+}
+
+impl<O: IsA<Clipboard>> ClipboardExtManual for O {
+    fn set_with_data<F: Fn(&Clipboard, &SelectionData, c_uint) + 'static>(&self, targets: &[TargetEntry], f: F) -> bool {
```

Comment:
I think we never directly expose C-like types (I'm referring to `c_uint` in here).

Reply:
OK, I will replace this with u32, which is the same type as in TargetEntry::info field.
Those two are supposed to go together.

Reply:
Use the `from_glib` function to make the conversion.

Reply:
@GuillaumeGomez you refer to u32 -> c_uint conversion? As far as I can see
there seems to be no suitable FromGlib implementation. Either way those types
are exactly the same on all platforms supported by Rust so far.

Reply:
I thought it did exist. It's been a while I didn't write bindings by hand so I'm maybe completely wrong...

cc @sdroege @EPashkin